### PR TITLE
Align subtree definition with latest specs

### PR DESCRIPTION
**Pull request by phbnf**
Updates subtree consistency proof generation, verification, and test vectors to align with recent changes in ietf-plants-wg/merkle-tree-certs:
 - [ietf-plants-wg/merkle-tree-certs - PR#275](https://github.com/ietf-plants-wg/merkle-tree-certs/pull/275)
 - [ietf-plants-wg/merkle-tree-certs - PR#274](https://github.com/ietf-plants-wg/merkle-tree-certs/pull/274)

Specifically:
  - **Empty subtrees:** `start == end` is now treated as a valid 0-length subtree. Subtree consistency proofs for empty subtrees return an empty array and verify that the subtree root equals the empty tree hash (`HASH()`), without depending on the parent tree's root.
  - **`FindSubtrees` edge case:** Intervals of size <= 1 now return two subtrees (`[start, end)` and `[end, end)`) instead of one. It now returns two subtrees instead of returning an array.
  - **Tests & Probes:** Updated test loops to include empty subtrees (`start == end`), updated reference vectors to match the new rolling SHA-256 digests, and added empty subtree test cases to `proofgen`. Take this as an opportunity to define `staticSubtreeConsistencyProbes` and `invalidSubtreeConsistencyProof` statically, rather than dynamically from consistency proof equivalents.

Towards #225

### CODE REVIEW

File: proof/proof.go [hunk 1]
```diff
@@ -273,24 +276,24 @@ type Subtree struct {
...
+//   - If the provided subtree has a size of 1, then this function returns that subtree and an empty subtree.
+//   - If the provided [start, end) range is already a valid subtree, then it is still split into two smaller subtrees.
+//   - The 2nd subtree, if not empty, is adjacent to the first, and may not be a perfect subtree.
+//   - The returned subtrees fully cover the [start, end) range.
 //   - There are no "extra" entries covered past end, but there may be covered entries prior to start.
 //   - The number of entries covered before start is always less than half the size of the first returned subtree.
-func FindSubtrees(start, end uint64) ([]Subtree, error) {
-	if start >= end {
-		return nil, fmt.Errorf("start %d must be strictly less than end %d", start, end)
+func FindSubtrees(start, end uint64) (Subtree, Subtree, error) {
```
Comment (AlCutter) [inline proof/proof.go]: Now that we now _always_ return two subtrees, _and_ those subtrees are guaranteed to be touching, we should ditch the `Subtree` struct and just return `start, mid, end`.

This makes it impossible to return non-adjacent ranges, and gets rid of the asymmetry where we return a struct representing subtrees here, but don't take the same subtree struct as an argument anywhere else.

**Review by AlCutter — APPROVED** [review]
